azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -261,6 +266,19 @@ def __get_all_agents_on_disk():
         path = os.path.join(conf.get_lib_dir(), "{0}-*".format(AGENT_NAME))
         return [GuestAgent.from_installed_agent(path=agent_dir) for agent_dir in glob.iglob(path) if os.path.isdir(agent_dir)]
 
+    def __check_if_downgrade_is_requested_and_allowed(self, requested_version):
+        """
+        Don't allow downgrades for self-update version
+        Note: The intention of this check is to keep the original behavior of self-update as it is.
+        """
+        if not self._is_requested_version_update:
+            if requested_version < CURRENT_VERSION:
+                msg = "Downgrade requested in the GoalState, but downgrades are not supported for self-update version:{0}, " \
+                      "skipping agent update".format(requested_version)
+                logger.verbose(msg)
```

Comment:
should be INFO and probably also send telemetry

Reply:
I thought we only hit this scenario in test, so this is not useful info for tests. In production, I doubt if this happens unless bug in CRP or PIR. This may be useful that time, but caveat is we log it every hour.

Reply:
every hour is ok